pkg/api/doctor/doctor.go

```diff
@@ -0,0 +1,51 @@
+package doctor
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/hospital-management/db"
+	"github.com/hospital-management/pkg/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+func CreateDoctor(w http.ResponseWriter, r *http.Request) {
+	var doc utils.Doctor
+	err := json.NewDecoder(r.Body).Decode(&doc)
+	if err != nil {
+		log.Println("error occured during decoding input body", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if doc.Name == "" || doc.MobileNo == 0 {
+		log.Println("invalid credentials")
```

Comment:
These are not credentials and also error message should have what attribute missed and that info should also be attached to response.